### PR TITLE
build: fix cross-arch macOS M1 build reproducibility

### PR DESCRIPTION
**Pull request by fanquake**
Building the macOS M1 bitcoin-qt binary at a optimisation level higher than `-O1` causes reproducibility issues when building on different architectures.

Proposing somewhat of a hammer.

This would fix 1 of the 2 remaining HOSTS in #21194.

Guix Build (x86_64):
```bash
1b58b5109b32dca2509499c93347148e6bab5dca835081f8cbd3123bed72cce1  guix-build-08bd3382777b/output/arm64-apple-darwin/SHA256SUMS.part
0e0d063d3832fad7c5116dabb2ac33c919f40bda04759aad4523c6247295bc9e  guix-build-08bd3382777b/output/arm64-apple-darwin/bitcoin-08bd3382777b-arm64-apple-darwin-unsigned.dmg
1feb301245f2c664edcfd9ac528fe1543fc7b183b3b42637db77d57658bc2b5e  guix-build-08bd3382777b/output/arm64-apple-darwin/bitcoin-08bd3382777b-arm64-apple-darwin-unsigned.tar.gz
641eb100d0a281203f9d6e36e45dc0ffc772c680d6aec462434f106b4c44e295  guix-build-08bd3382777b/output/arm64-apple-darwin/bitcoin-08bd3382777b-arm64-apple-darwin.tar.gz
9d89920626e35939aa6cf506fc85861179f3c0e18d4ef1954750cf81336a851a  guix-build-08bd3382777b/output/dist-archive/bitcoin-08bd3382777b.tar.gz
e7697d30084270d0b5843b3baf0d752e240c2f708f728bc2f6896f153276ca6b  guix-build-08bd3382777b/output/x86_64-apple-darwin/SHA256SUMS.part
dd77acee082dbfd3cdad2c564bbd3bdace8df9bf32f92cf4a2debd5a996ace49  guix-build-08bd3382777b/output/x86_64-apple-darwin/bitcoin-08bd3382777b-x86_64-apple-darwin-unsigned.dmg
85e63fccb7af12468a04a678034c42dcd775d243b2d194a52e1086a6ffbdbe84  guix-build-08bd3382777b/output/x86_64-apple-darwin/bitcoin-08bd3382777b-x86_64-apple-darwin-unsigned.tar.gz
ff2629957608898d76a42025985e3ec4bf5dc8572794e32b4182ba6f8babb828  guix-build-08bd3382777b/output/x86_64-apple-darwin/bitcoin-08bd3382777b-x86_64-apple-darwin.tar.gz
```

Guix Build (arm64):
```bash
1b58b5109b32dca2509499c93347148e6bab5dca835081f8cbd3123bed72cce1  guix-build-08bd3382777b/output/arm64-apple-darwin/SHA256SUMS.part
0e0d063d3832fad7c5116dabb2ac33c919f40bda04759aad4523c6247295bc9e  guix-build-08bd3382777b/output/arm64-apple-darwin/bitcoin-08bd3382777b-arm64-apple-darwin-unsigned.dmg
1feb301245f2c664edcfd9ac528fe1543fc7b183b3b42637db77d57658bc2b5e  guix-build-08bd3382777b/output/arm64-apple-darwin/bitcoin-08bd3382777b-arm64-apple-darwin-unsigned.tar.gz
641eb100d0a281203f9d6e36e45dc0ffc772c680d6aec462434f106b4c44e295  guix-build-08bd3382777b/output/arm64-apple-darwin/bitcoin-08bd3382777b-arm64-apple-darwin.tar.gz
9d89920626e35939aa6cf506fc85861179f3c0e18d4ef1954750cf81336a851a  guix-build-08bd3382777b/output/dist-archive/bitcoin-08bd3382777b.tar.gz
e7697d30084270d0b5843b3baf0d752e240c2f708f728bc2f6896f153276ca6b  guix-build-08bd3382777b/output/x86_64-apple-darwin/SHA256SUMS.part
dd77acee082dbfd3cdad2c564bbd3bdace8df9bf32f92cf4a2debd5a996ace49  guix-build-08bd3382777b/output/x86_64-apple-darwin/bitcoin-08bd3382777b-x86_64-apple-darwin-unsigned.dmg
85e63fccb7af12468a04a678034c42dcd775d243b2d194a52e1086a6ffbdbe84  guix-build-08bd3382777b/output/x86_64-apple-darwin/bitcoin-08bd3382777b-x86_64-apple-darwin-unsigned.tar.gz
ff2629957608898d76a42025985e3ec4bf5dc8572794e32b4182ba6f8babb828  guix-build-08bd3382777b/output/x86_64-apple-darwin/bitcoin-08bd3382777b-x86_64-apple-darwin.tar.gz
```

### CODE REVIEW

**Comment by hebasto** [comment]
> Proposing somewhat of a hammer.

Indeed :smile:

**Comment by achow101** [comment]
x86_64 build:

```b6b7395fd0d67c32894b520ef6300d467b3c5ddc52733efd81243079e45c9184  guix-build-849a1f5c7b77/output/arm64-apple-darwin/SHA256SUMS.part
a6c5bc5af6992935ad4a547f471e18dbf5419b3284f95a78e65fc3a0c491cfe4  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.dmg
bf4f3cbb5ea2559251c6cca95205599456b6e597f32ccdf2a1d93a4384a932e0  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.tar.gz
f5118d115371a361c24d42af3b4be97d1fe340f0f19300290b74805106048a8a  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin.tar.gz
29ae09d6b2d52f814bead74dc1a51a5d4944af29a38fa97acde3ca68ad83ecc9  guix-build-849a1f5c7b77/output/dist-archive/bitcoin-849a1f5c7b77.tar.gz
e86da978c62ac1ad5fe0df8a61c98e86203dbf4fd7ec7619d07af2ad957b9ef4  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/SHA256SUMS.part
d6433cf54052308bd632c2f093260d355c51386e6751977a5a25a60cad14365e  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.dmg
d94dc7e2c4da8b45a36c39b35dda48b960e7c05a59834825215eb64daf5dcf06  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.tar.gz
ccda72b619d9bf88c0baca11875ea94091080befc1e19e4031778919ce5f5be1  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin.tar.gz
```

arm64 build:
```
b6b7395fd0d67c32894b520ef6300d467b3c5ddc52733efd81243079e45c9184  guix-build-849a1f5c7b77/output/arm64-apple-darwin/SHA256SUMS.part
a6c5bc5af6992935ad4a547f471e18dbf5419b3284f95a78e65fc3a0c491cfe4  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.dmg
bf4f3cbb5ea2559251c6cca95205599456b6e597f32ccdf2a1d93a4384a932e0  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.tar.gz
f5118d115371a361c24d42af3b4be97d1fe340f0f19300290b74805106048a8a  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin.tar.gz
29ae09d6b2d52f814bead74dc1a51a5d4944af29a38fa97acde3ca68ad83ecc9  guix-build-849a1f5c7b77/output/dist-archive/bitcoin-849a1f5c7b77.tar.gz
e86da978c62ac1ad5fe0df8a61c98e86203dbf4fd7ec7619d07af2ad957b9ef4  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/SHA256SUMS.part
d6433cf54052308bd632c2f093260d355c51386e6751977a5a25a60cad14365e  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.dmg
d94dc7e2c4da8b45a36c39b35dda48b960e7c05a59834825215eb64daf5dcf06  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.tar.gz
ccda72b619d9bf88c0baca11875ea94091080befc1e19e4031778919ce5f5be1  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin.tar.gz
```

**Comment by jarolrod** [comment]
guix hashes

x86:
```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum

b6b7395fd0d67c32894b520ef6300d467b3c5ddc52733efd81243079e45c9184  guix-build-849a1f5c7b77/output/arm64-apple-darwin/SHA256SUMS.part
a6c5bc5af6992935ad4a547f471e18dbf5419b3284f95a78e65fc3a0c491cfe4  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.dmg
bf4f3cbb5ea2559251c6cca95205599456b6e597f32ccdf2a1d93a4384a932e0  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.tar.gz
f5118d115371a361c24d42af3b4be97d1fe340f0f19300290b74805106048a8a  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin.tar.gz
29ae09d6b2d52f814bead74dc1a51a5d4944af29a38fa97acde3ca68ad83ecc9  guix-build-849a1f5c7b77/output/dist-archive/bitcoin-849a1f5c7b77.tar.gz
e86da978c62ac1ad5fe0df8a61c98e86203dbf4fd7ec7619d07af2ad957b9ef4  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/SHA256SUMS.part
d6433cf54052308bd632c2f093260d355c51386e6751977a5a25a60cad14365e  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.dmg
d94dc7e2c4da8b45a36c39b35dda48b960e7c05a59834825215eb64daf5dcf06  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.tar.gz
ccda72b619d9bf88c0baca11875ea94091080befc1e19e4031778919ce5f5be1  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin.tar.gz
```

arm64:
```

```

**Comment by hebasto** [comment]
For the reference, currently the `corelib` and `gui` libraries are being compiled with `QMAKE_CFLAGS_OPTIMIZE_FULL = -O3`.

Other libraries (`network`,  `widgets`, `plugins` and `testlib`) are being compiled with `QMAKE_CFLAGS_OPTIMIZE = -O2`.

**Comment by hebasto** [comment]
Guix builds on `arm64`:
```
b6b7395fd0d67c32894b520ef6300d467b3c5ddc52733efd81243079e45c9184  guix-build-849a1f5c7b77/output/arm64-apple-darwin/SHA256SUMS.part
a6c5bc5af6992935ad4a547f471e18dbf5419b3284f95a78e65fc3a0c491cfe4  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.dmg
bf4f3cbb5ea2559251c6cca95205599456b6e597f32ccdf2a1d93a4384a932e0  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin-unsigned.tar.gz
f5118d115371a361c24d42af3b4be97d1fe340f0f19300290b74805106048a8a  guix-build-849a1f5c7b77/output/arm64-apple-darwin/bitcoin-849a1f5c7b77-arm64-apple-darwin.tar.gz
29ae09d6b2d52f814bead74dc1a51a5d4944af29a38fa97acde3ca68ad83ecc9  guix-build-849a1f5c7b77/output/dist-archive/bitcoin-849a1f5c7b77.tar.gz
e86da978c62ac1ad5fe0df8a61c98e86203dbf4fd7ec7619d07af2ad957b9ef4  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/SHA256SUMS.part
d6433cf54052308bd632c2f093260d355c51386e6751977a5a25a60cad14365e  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.dmg
d94dc7e2c4da8b45a36c39b35dda48b960e7c05a59834825215eb64daf5dcf06  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin-unsigned.tar.gz
ccda72b619d9bf88c0baca11875ea94091080befc1e19e4031778919ce5f5be1  guix-build-849a1f5c7b77/output/x86_64-apple-darwin/bitcoin-849a1f5c7b77-x86_64-apple-darwin.tar.gz
```

File: depends/packages/qt.mk [hunk 1]
```diff
@@ -131,6 +131,9 @@ $(package)_config_opts_darwin += -no-feature-corewlan
 $(package)_config_opts_darwin += -no-freetype
 $(package)_config_opts_darwin += QMAKE_MACOSX_DEPLOYMENT_TARGET=$(OSX_MIN_VERSION)
 
+$(package)_config_opts_aarch64_darwin += "QMAKE_CFLAGS_OPTIMIZE = '-O1'"
+$(package)_config_opts_aarch64_darwin += "QMAKE_CFLAGS_OPTIMIZE_FULL = '-O1'"
+
```
Comment (hebasto) [inline depends/packages/qt.mk:136]:
```suggestion
$(package)_config_opts_aarch64_darwin += "QMAKE_CFLAGS_OPTIMIZE_FULL = -O1"

```

will be enough. It changes optimization from `-O3` to `-O1` for the `corelib` and `gui` libraries only. Actually, we are interested in the latter only.

Comment (hebasto) [reply depends/packages/qt.mk:136]: Also it would be nice to have a comment here.

Comment (fanquake) [reply depends/packages/qt.mk]: Adjusted and added a comment.

**Review by hebasto — APPROVED** [review]
ACK 08bd3382777bf5d24fa8f32f27abc7b6d1e20dd9

**Review by jarolrod — COMMENTED** [review]
ACK 08bd3382777bf5d24fa8f32f27abc7b6d1e20dd9

tested the resulting binary

**Comment by Kingmax1981** [comment]
Many

**Review by Kingmax1981 — COMMENTED** [review]
Many

**Pull request merged**